src/main/java/Group15/View/MuscleSelectionView.java

```diff
@@ -155,35 +157,62 @@ private static ScrollPane createEquipmentSelectorScrollPane() {
             index++;
         }
 
-        createEquipmentCheckboxFunctionality(allCheckbox, equipmentCheckBoxes);
+        createEquipmentCheckboxFunctionality(selectAllButton, deselectAllButton, equipmentCheckBoxes);
 
         ScrollPane equipmentSelectorScrollPane = new ScrollPane(equipmentSelectorGridPane);
         equipmentSelectorScrollPane.setFitToWidth(true);
-        equipmentSelectorScrollPane.setMinSize(350, 100);
+        equipmentSelectorScrollPane.setMinSize(350, 140);
 
         return equipmentSelectorScrollPane;
     }
 
-    private static void createEquipmentCheckboxFunctionality(CheckBox allCheckbox, List<CheckBox> equipmentCheckBoxes) {
-        allCheckbox.setOnAction(_ -> {
-            if (allCheckbox.isSelected()) {
-                MuscleSelectionView.selectedEquipment.clear();
-                for (CheckBox checkbox : equipmentCheckBoxes) {
-                    if (checkbox != allCheckbox) checkbox.setSelected(false);
+    private static void createEquipmentCheckboxFunctionality(Button allButton, Button noneButton, List<CheckBox> equipmentCheckBoxes) {
+        allButton.setOnAction(_ -> {
+            for (Equipment equipment : Equipment.values()) {
+                if(!selectedEquipment.contains(equipment)) {
+                    MuscleSelectionView.selectedEquipment.add(equipment);
+
                 }
             }
+            for (CheckBox checkBox : equipmentCheckBoxes) {
+                checkBox.setSelected(true);
+            }
         });
-        for (CheckBox checkbox : equipmentCheckBoxes) {
-            if (checkbox != allCheckbox) {
-                checkbox.setOnAction(_ -> {
-                    if (checkbox.isSelected()) {
-                        allCheckbox.setSelected(false);
-                        MuscleSelectionView.selectedEquipment.remove(Equipment.valueOf(checkbox.getText()));
-                        if (!MuscleSelectionView.selectedEquipment.contains(Equipment.valueOf(checkbox.getText()))) MuscleSelectionView.selectedEquipment.add(Equipment.valueOf(checkbox.getText()));
-                    } else {
-                        MuscleSelectionView.selectedEquipment.remove(Equipment.valueOf(checkbox.getText()));
-                    }
-                });
+
+        noneButton.setOnAction(_ -> {
+            for (Equipment equipment : Equipment.values()) {
+                if(selectedEquipment.contains(equipment)) {
+                    MuscleSelectionView.selectedEquipment.remove(equipment);
+
+                }
+            }
+            for (CheckBox checkBox : equipmentCheckBoxes) {
+                checkBox.setSelected(false);
+            }
+        });
+
+        for (CheckBox checkBox : equipmentCheckBoxes) {
+            checkBox.setOnAction(e -> {
+                if (e.getSource() instanceof CheckBox) {
+                    handleCheckBoxAction((CheckBox) e.getSource());
```

Comment:
This is not what I meant, this does not add anything to what I commented last time